src/components/shapes/defaults.js

```diff
@@ -30,18 +30,22 @@ function handleShapeDefaults(shapeIn, shapeOut, fullLayout) {
     }
 
     var visible = coerce('visible');
-
     if(!visible) return;
 
+    var dfltType = shapeIn.path ? 'path' : 'rect';
+    var shapeType = coerce('type', dfltType);
```

Comment:
hmm not new here, just rearranged, but template support... may be better to do
```js
path = coerce('path');
var dfltType = path ? 'path' : 'rect';
var shapeType = coerce('type', dfltType);
if (type !== 'path') delete shapeOut.path;
```

Reply:
Good eye! Thanks for the hint.
Addressed in https://github.com/plotly/plotly.js/pull/4775/commits/092a80e5976566ab15211e14926affb9ca99033b.